iambic/plugins/v0_1_0/aws/organizations/scp/utils.py

```diff
@@ -0,0 +1,572 @@
+            "Targets": targets[0],
+            "PolicyDocument": policy.get("Content", {}),
+            "Tags": tags[0],
+        },
+    )
+
+
+async def detach_policy(client, policyId, targetId):
+    log.info(f"Detaching policy {policyId} from {targetId}")
+    await boto_crud_call(
+        client.detach_policy,
+        PolicyId=policyId,
+        TargetId=targetId,
+    )
+    log.info(f"Detached policy {policyId} from {targetId}")
+
+
+async def delete_policy(client, policyId: str, *args, **kwargs):
+    """
+    Deletes the specified policy from your organization.
+    Before you perform this operation, you must first detach
+    the policy from all organizational units (OUs), roots, and accounts.
+    """
+    list_targets_for_policy_semaphore = NoqSemaphore(list_targets_for_policy, 10)
+
+    targets = chain.from_iterable(
+        await list_targets_for_policy_semaphore.process(
+            [{"client": client, "policyId": policyId}]
+        )
+    )
+
+    targets_tasks = [
+        detach_policy(client, policyId, target.TargetId) for target in targets
+    ]
+
+    await asyncio.gather(*targets_tasks)
+
+    log.info(f"Deleting policy {policyId}")
+
+    await boto_crud_call(client.delete_policy, PolicyId=policyId)
+
+    log.info(f"Deleted policy {policyId}")
+
+
+async def create_policy(client, policy):
+    if isinstance(policy["PolicyDocument"], dict):
+        policy["PolicyDocument"] = json.dumps(policy["PolicyDocument"])
+
+    res = await boto_crud_call(
+        client.create_policy,
+        Content=policy["PolicyDocument"],
+        Description=policy.get("Description", ""),
+        Name=policy.get("PolicyName", f"NewPolicy-{random.randint(0, 100):03d}"),
+        Type="SERVICE_CONTROL_POLICY",
+        Tags=policy.get("Tags", []),
+    )
+
+    policy.update(
+        PolicyId=res.get("Policy", {}).get("PolicySummary", {}).get("Id"),
+        PolicyName=res.get("Policy", {}).get("PolicySummary", {}).get("Name"),
+    )
+
+    return res.get("Policy", {}).get("PolicySummary", {})
+
+
+async def describe_organization(
+    client,
+):
+    """Retrieves information about the organization that the user's account belongs to.
+    This operation can be called from any account in the organization.
+    """
+    res = await boto_crud_call(client.describe_organization)
+    return res.get("Organization", {})
+
+
+async def service_control_policy_is_enabled(client):
+    """Check if SCPs are enabled for the organization."""
+    org = await describe_organization(client)
+
+    return (
+        len(
+            [
+                apt
+                for apt in org.get("AvailablePolicyTypes", [])
+                if apt.get("Type") == "SERVICE_CONTROL_POLICY"
+                and apt.get("Status") == "ENABLED"
+            ]
+        )
+        > 0
+    )
+
+
+async def apply_update_policy(
+    client,
+    policy,
+    current_policy,
+    log_params,
+    *args,
+) -> list[ProposedChange]:
+    current_value = dict(
```

Comment:
I see here it's .get("Name") but on line 249 it's .get("PolicyName"). Does that difference matter?

Reply:
`policy` is fill by our template file properties. In that place we have "policy_name, policy_id & policy_document"
`current_policy`is fill by boto3 response (when we executed get_policy by id). The structure is not the same. 

Here, I tried to not use the pydantic models (if you go one step outside, you will find something like "current_policy = current_policy.dict()").